src/main/java/io/jenkins/plugins/synopsys/security/scan/extension/freestyle/SecurityScanFreestyle.java

```diff
@@ -0,0 +1,428 @@
+            throws PluginExceptionHandler {
+        return ScanParametersFactory.preparePipelineParametersMap(
+                this, ScanParametersFactory.getGlobalConfigurationValues(workspace, listener), listener);
+    }
+
+    @Override
+    public void perform(
+            @NonNull Run<?, ?> run,
+            @NonNull FilePath workspace,
+            @NonNull EnvVars env,
+            @NonNull Launcher launcher,
+            @NonNull TaskListener listener) {
+
+        listener.getLogger()
+                .println(
+                        "**************************** START EXECUTION OF SYNOPSYS SECURITY SCAN ****************************");
+
+        try {
+            ScanParametersFactory.createPipelineCommand(run, listener, env, launcher, null, workspace)
+                    .initializeScanner(getParametersMap(workspace, listener));
+        } catch (Exception e) {
+            if (e instanceof PluginExceptionHandler) {
+                try {
+                    throw new PluginExceptionHandler("Workflow failed! " + e.getMessage());
+                } catch (PluginExceptionHandler ex) {
+                    throw new RuntimeException(ex);
+                }
+            } else {
+                try {
+                    throw new ScannerException(ExceptionMessages.scannerFailureMessage(e.getMessage()));
+                } catch (ScannerException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }
+        } finally {
+            listener.getLogger()
+                    .println(
+                            "**************************** END EXECUTION OF SYNOPSYS SECURITY SCAN ****************************");
+        }
+    }
+
+    @Extension
+    public static class Descriptor extends BuildStepDescriptor<Builder> {
+
+        @Override
+        public String getDisplayName() {
+            return "Synopsys Security Scan";
+        }
+
+        @Override
+        public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+            return jobType.isAssignableFrom(FreeStyleProject.class);
+        }
+
+        @SuppressWarnings({"lgtm[jenkins/no-permission-check]", "lgtm[jenkins/csrf]"})
+        public ListBoxModel doFillProductItems() {
+            ListBoxModel items = new ListBoxModel();
+            Map<String, String> customLabels = new HashMap<>();
+
+            items.add(new ListBoxModel.Option("Select", "select"));
+            customLabels.put(SecurityProduct.BLACKDUCK.name().toLowerCase(), "Black Duck");
+            customLabels.put(SecurityProduct.COVERITY.name().toLowerCase(), "Coverity");
+            customLabels.put(SecurityProduct.POLARIS.name().toLowerCase(), "Polaris");
+
+            for (SecurityProduct product : SecurityProduct.values()) {
+                String value = product.name().toLowerCase();
+                String label = customLabels.getOrDefault(value, product.name());
```

Comment:
Instead of preparing Map and using the Map here, we can have corresponding field in SecurityProduct enum with values "Black Duck", "Coverity" and "Polaris".

Reply:
Yeah. Thanks for your suggestion.